lib/vmdb/settings.rb

```diff
@@ -198,5 +206,34 @@ def self.apply_settings_changes(resource, deltas)
       end
     end
     private_class_method :apply_settings_changes
+
+    def self.process_magic_value(resource, delta)
+      return false unless MAGIC_VALUES.include?(delta[:value])
+      case delta[:value]
+      when DELETE_COMMAND
+        _log.info("removing custom settings #{delta[:key]} on #{resource.class} id:#{resource.id}")
+        resource.settings_changes.where("key LIKE ?", "#{delta[:key]}%").destroy_all
+      when DELETE_ALL_COMMAND
+        _log.info("resetting #{delta[:key]} to defaul for all resorces")
+        SettingsChange.where("key LIKE ?", "#{delta[:key]}%").destroy_all
+      end
+      true
+    end
+    private_class_method :process_magic_value
+
+    def self.walk_hash(hash, path = [], &block)
```

Comment:
I'm not sure I understand why we are keeping track of this `path` parameter here.
We are yielding it and passing it along into the next level of recursion, but don't use it in the block in `.remove_magic_values`

Reply:
you are right, we do not need it here if we are ignoring Arrays
I was thinking to process Arrays too (as in ```SettingsWalker``` from were this method was copied), but could not find example in our ```settings.yml``` and decided not top do so.
Removing it ...

@Fryguy should we accommodate resetting nested Hash as an element of Array in `Settings` ?